internal/exec/stages/files/filesystemEntries.go

```diff
@@ -393,6 +393,9 @@ func (s stage) getOrderedCreationList(config types.Config) ([]filesystemEntry, e
 	// Map from paths in the config to where they resolve for duplicate checking
 	paths := map[string]string{}
 	for _, d := range config.Storage.Directories {
+		if d.Mode != nil {
+			d.Mode = cutil.IntToPtr(int(util.ToFileMode(d.Mode)))
```

Comment:
Changing the semantics of data midflight like this is dangerous, since it's easy to have code paths that accidentally don't translate (or translate twice, etc.).  It's much better to have e.g. a well-defined struct translation step that produces a different datatype, so the compiler will detect mismatches.  We don't have an easy opportunity for that here.  However, nothing in the codebase uses the `ModeSetuid` etc. flags, and none of the [FileMode flags](https://pkg.go.dev/io/fs#FileMode) collide with 07000, so I think the translation can be deferred all the way to `Util.SetPermissions`.